core/managementserver/server.go

```diff
@@ -35,8 +39,13 @@ func New() *Server {
 
 // GetKeySet returns a list of public keys (a keyset) that corresponds to the signing keys
 // this service has for a given domain and app.
-func (s *Server) GetKeySet(context.Context, *pb.GetKeySetRequest) (*tpb.KeySet, error) {
-	return nil, status.Errorf(codes.Unimplemented, "unimplemented")
+func (s *Server) GetKeySet(ctx context.Context, in *pb.GetKeySetRequest) (*tpb.KeySet, error) {
+	ks, err := s.keysets.Get(ctx, s.instance, in.GetDomainId(), in.GetAppId())
+	if err != nil {
+		return nil, err
+	}
+	ks.SigningKeys = nil // DON'T LEAK PRIVATE KEYS!!
```

Comment:
Great comment!  This is all fine as long as the comment remains and there's just this simple single path through the func to something being returned.

I wonder if instead the sanitization could be done within `storage.KeySets.Get()`.

You'd need an additional API though, as presumably some legitimate `Get()` calls do want to retrieve the private key. The additional API could have a much more explicit name like `GetWithPrivate()` or something.

Reply:
I'm hoping to replace KeySets with [Tink](https://github.com/google/tink), which should remove this particular problem.